tests/components/AddonList.test.js

```diff
@@ -35,44 +36,105 @@ describe('<AddonList />', () => {
             "deployed.owa.name": null
         },
         install: false
-    }];
+    },
+     {"appDetails":
+       {
+         "uuid":"atlas",
+         "display":"Atlas Module",
+         "name":"Atlas Module",
+         "description":"The Atlas Module provides a mechanism for an implementation to create and manage a bubble on the\n\t\tOpenMRS Atlas (https://atlas.openmrs.org) directly from their OpenMRS server.",
+         "packageName":"org.openmrs.module.atlas",
+         "author":"Victor Chircu",
+         "version":"2.2",
+         "started":true,
+         "startupErrorMessage":null,
+         "requireOpenmrsVersion":"1.7.4",
+         "awareOfModules":["org.openmrs.module.legacyui"],
+         "requiredModules":["org.openmrs.module.uiframework"],
+         "links":[
+           {
+             "rel":"ref",
+             "uri":"http://localhost:8081/openmrs-standalone/ws/rest/v1/module/atlas?v=ref"},
+           {
+             "rel":"action",
+             "uri":"http://localhost:8081/openmrs-standalone/ws/rest/v1/moduleaction"},
+           {
+             "rel":"self",
+             "uri":"http://localhost:8081/openmrs-standalone/ws/rest/v1/module/atlas"}],
+         "resourceVersion":"1.8"
+       },
+       "appType":"module",
+       "install":false
+     }
+  ];
+
 
     const addonList = [];
     const searchedAddons = [];
     const openPage = () => { };
     const openModal = () => { };
     const handleDownload = () => { };
+    const getInstalledall = () => { };
+    const handleUserClick = () => { };
+    const handleInstall = () => { };
+    const handleUpgrade = () => { };
     const getInstalled = () => { };
+    const getInstalledStub = stub().onCall(0).returns(appList);
     const testAddonList = mount(
         <AddonList
             addonList={appList}
             searchedAddons={searchedAddons}
             openPage={openPage}
             openModal={openModal}
             handleDownload={handleDownload}
+            handleUserClick={handleUserClick}
+            handleInstall={handleInstall}
+            handleUpgrade={handleUpgrade}
             getInstalled={getInstalled} />);
 
+    const searchedAddonslist = mount(
```

Comment:
You should use `shallow` instead of `mount` unless you want to test componentDidMount, componentDidUpdate, component lifecycle or children behavior.

Reply:
I want it to do that @patlub  since the component is receiving for props from a parent component and I would want it to update when it receives the props that why I opted to use `mount` rather than shallow.

Reply:
Okay.